MaxMind.GeoIP2/WebServiceClient.cs

```diff
@@ -297,24 +297,28 @@ public InsightsResponse Insights(string ipAddress)
         /// <returns>An <see cref="InsightsResponse" /></returns>
         public InsightsResponse Insights()
         {
-            return Insights((IPAddress)null);
+            return Execute<InsightsResponse>("insights", null);
         }
 
 #endif
 
         private static IPAddress ParseIP(string ipAddress)
         {
-            IPAddress ip = null;
+            IPAddress? ip = null;
+
+            // The "ipAddress != null" is here for backwards compatibility with
+            // pre-nullable-reference-types code that might possible rely on the
+            // undocumented feature of passing a null IP string.
             if (ipAddress != null && !IPAddress.TryParse(ipAddress, out ip))
             {
                 throw new GeoIP2Exception($"The specified IP address was incorrectly formatted: {ipAddress}");
             }
-            return ip;
+            return ip!;

```

Comment:
If ipAddress was null though, will this be null? Or doesn't matter because it'd be code not using the NRT stuff?

Reply:
Right, this is what the comment above is about. There is a chance that someone using the API currently is taking advantage of the undocumented ability to pass a null IP address to those methods, but I don't want to make that part of the API (and, even if I did, it would make it hard to reconcile with the interface that is shared with the database reader).

If someone is using "null reference types" and hasn't disabled the check, the "!" should be true. I am happy to improve the explanation if you have suggestions on a wording that might help.